From 5a393f6ef54b80df0c3d46a1ca7f3a75cd9a29a8 Mon Sep 17 00:00:00 2001
From: Ville Savolainen <ville.savolainen@open-xchange.com>
Date: Wed, 3 Jun 2026 09:43:20 +0300
Subject: [PATCH] global: Fix Lua linking issue by separating libs from
 dependencies

In src/auth/Makefile.am, LUA_LIBS was incorrectly added to auth_libs, which is used in auth_DEPENDENCIES. This caused make to fail when LUA_LIBS contained linker flags like -L (e.g. on Wolfi OS).

Also moved LUA_LIBS from LDFLAGS to LIBADD in mail-lua plugin.
---
 src/auth/Makefile.am             | 18 ++++++------------
 src/plugins/mail-lua/Makefile.am |  5 +++--
 2 files changed, 9 insertions(+), 14 deletions(-)

diff --git a/src/auth/Makefile.am b/src/auth/Makefile.am
index 7474e76fb68..d0a25c046ae 100644
--- a/src/auth/Makefile.am
+++ b/src/auth/Makefile.am
@@ -21,12 +21,10 @@ LUA_LIB =
 AUTH_LUA_LIBS =
 AUTH_LUA_LDADD =
 if HAVE_LUA
-
-if AUTH_LUA_PLUGIN
-LUA_LIB += libauthdb_lua.la
-else
 AUTH_LUA_LIBS += $(LIBDOVECOT_LUA)
 AUTH_LUA_LDADD += $(LUA_LIBS)
+if AUTH_LUA_PLUGIN
+LUA_LIB += libauthdb_lua.la
 endif
 endif
 
@@ -199,10 +197,6 @@ auth_libs += $(LIBDOVECOT_LDAP)
 endif
 endif
 
-if HAVE_LUA
-auth_libs += $(LIBDOVECOT_LUA) $(LUA_LIBS)
-endif
-
 if AUTH_LUA_PLUGIN
 libauthdb_lua_la_LDFLAGS = -module -avoid-version -shared
 libauthdb_lua_la_LIBADD = $(LIBDOVECOT_LUA)
@@ -249,7 +243,7 @@ test_auth_SOURCES = \
 	test-mock.c \
 	test-main.c
 
-test_auth_LDADD = $(test_auth_ldadd_plugins) $(LIBDOVECOT) $(auth_libs) $(AUTH_LIBS)
+test_auth_LDADD = $(test_auth_ldadd_plugins) $(LIBDOVECOT) $(auth_libs) $(AUTH_LIBS) $(AUTH_LUA_LDADD)
 test_auth_DEPENDENCIES = $(test_auth_ldadd_plugins) $(pkglibexec_PROGRAMS) $(LIBDOVECOT_DEPS)
 
 test_mech_SOURCES = \
@@ -258,7 +252,7 @@ test_mech_SOURCES = \
 	test-mock.c \
 	test-mech.c
 
-test_mech_LDADD = $(LIBDOVECOT) $(auth_libs) $(AUTH_LIBS)
+test_mech_LDADD = $(LIBDOVECOT) $(auth_libs) $(AUTH_LIBS) $(AUTH_LUA_LDADD)
 test_mech_DEPENDENCIES = $(pkglibexec_PROGRAMS) $(LIBDOVECOT_DEPS)
 
 test_auth_client_SOURCES = \
@@ -267,7 +261,7 @@ test_auth_client_SOURCES = \
 	test-mock.c \
 	test-auth-client.c
 
-test_auth_client_LDADD = $(LIBDOVECOT) $(auth_libs) $(AUTH_LIBS)
+test_auth_client_LDADD = $(LIBDOVECOT) $(auth_libs) $(AUTH_LIBS) $(AUTH_LUA_LDADD)
 test_auth_client_DEPENDENCIES = $(pkglibexec_PROGRAMS) $(LIBDOVECOT_DEPS)
 
 test_auth_master_SOURCES = \
@@ -277,5 +271,5 @@ test_auth_master_SOURCES = \
 	test-auth-master.c \
 	test-auth-master-server.c
 
-test_auth_master_LDADD = $(LIBDOVECOT) $(auth_libs) $(AUTH_LIBS)
+test_auth_master_LDADD = $(LIBDOVECOT) $(auth_libs) $(AUTH_LIBS) $(AUTH_LUA_LDADD)
 test_auth_master_DEPENDENCIES = $(pkglibexec_PROGRAMS) $(LIBDOVECOT_DEPS)
diff --git a/src/plugins/mail-lua/Makefile.am b/src/plugins/mail-lua/Makefile.am
index d192c09bbe0..7c9c47b9db5 100644
--- a/src/plugins/mail-lua/Makefile.am
+++ b/src/plugins/mail-lua/Makefile.am
@@ -16,10 +16,11 @@ NOPLUGIN_LDFLAGS =
 module_LTLIBRARIES = \
 	lib01_mail_lua_plugin.la
 
-lib01_mail_lua_plugin_la_LDFLAGS = -module -avoid-version $(LUA_LIBS)
+lib01_mail_lua_plugin_la_LDFLAGS = -module -avoid-version
 lib01_mail_lua_plugin_la_LIBADD = \
 	../../lib-storage-lua/libdovecot-storage-lua.la \
-	$(LIBDOVECOT)
+	$(LIBDOVECOT) \
+	$(LUA_LIBS)
 lib01_mail_lua_plugin_la_SOURCES = \
 	mail-lua-plugin.c \
 	mail-lua-settings.c
